src/modules/storage/LedgerStorage.ts

```diff
@@ -3532,6 +3573,65 @@ export class LedgerStorage extends Storages {
         });
     }
 
+    /**
+     * Get proposal list
+     * @param limit Maximum record count that can be obtained from one query
+     * @param page The number on the page, this value begins with 1
+     * @returns returns the Promise with requested data
+     * and if an error occurs the .catch is called with an error.
+     */
+    public getProposals(limit: number, page: number): Promise<any[]> {
+        const sql = `
+                SELECT 
+                    P.proposal_id,
+                    P.proposal_title,
+                    P.proposal_type,
+                    P.fund_amount,
+                    P.vote_start_height,
+                    P.vote_end_height,
+                    P.proposal_status,
+                    M.submit_time,
+                    M.proposer_name,
+                    count(*) OVER() AS full_count
+                    FROM proposal P
+                    INNER JOIN proposal_metadata M
```

Comment:
Why is use INNER JOIN?
Proposal_metadata data may not exist.

Reply:
We need to display data from proposal_metadata table, the data is for frontend UI.